### DISCUSSION TITLE
Guided Exercise - HTML - Fundamentals of HTML

### DISCUSSION THREAD

**Discussion by akash-coded** [akash-coded/mern#229] [Tasks]
Since HTML is the backbone of web pages, understanding its structure and functionality is crucial. Here’s a step-by-step guided exercise that covers the fundamentals of HTML:

### Exercise: Building a Basic Web Page

**Objective:** 
Learn the basic structure of an HTML document by creating a simple web page.

#### Step 1: Setting Up Your Environment
1. **Install a Code Editor:** Download and install a code editor like Visual Studio Code or Sublime Text.
2. **Create a Project Folder:** Make a new folder on your computer named `html_basics`.
3. **Create an HTML File:** Inside this folder, create a new file named `index.html`.

#### Step 2: Basic Structure of an HTML Document
1. **Doctype Declaration:** At the top of your `index.html` file, type `<!DOCTYPE html>`. This tells the browser that this is an HTML5 document.
2. **HTML Element:** Below the doctype, add the `<html>` tag. This encompasses all your HTML code.
   ```html
   <!DOCTYPE html>
   <html>
   </html>
   ```

#### Step 3: Head and Body Sections
1. **Head Section:** Inside the `<html>` tag, add the `<head>` element. This section contains meta-information about the document.
   ```html
   <head>
       <title>My First HTML Page</title>
   </head>
   ```
2. **Body Section:** After the `<head>` section, add the `<body>` element. This is where the content visible to users will go.
   ```html
   <body>
   </body>
   ```

#### Step 4: Adding Content to the Body
1. **Heading:** Inside the `<body>`, add a heading using the `<h1>` tag: `<h1>Welcome to HTML Basics</h1>`.
2. **Paragraph:** Add a paragraph below the heading using the `<p>` tag.
   ```html
   <p>This is my first HTML page. I'm learning HTML!</p>
   ```

#### Step 5: Creating a List
1. **Unordered List:** Below the paragraph, create an unordered list using the `<ul>` tag.
2. **List Items:** Add items to the list using `<li>` tags.
   ```html
   <ul>
       <li>HTML Basics</li>
       <li>Introduction to CSS</li>
       <li>Getting Started with JavaScript</li>
   </ul>
   ```

#### Step 6: Adding Links and Images
1. **Hyperlink:** Add a link using the `<a>` tag. Set the `href` attribute to an external URL.
   ```html
   <a href="https://example.com">Visit Example.com</a>
   ```
2. **Image:** Add an image using the `<img>` tag. Set the `src` attribute to the image URL and add an `alt` description.
   ```html
   <img src="image_url.jpg" alt="Descriptive Text">
   ```

#### Step 7: Viewing Your Web Page
1. **Open in a Browser:** Open your `index.html` file in a web browser to see your created web page.

#### Step 8: Experimentation and Practice
1. **Modify Content:** Try changing the text in headings and paragraphs.
2. **Add More Elements:** Experiment by adding more HTML elements like `<div>`, `<span>`, and `<table>`.
3. **Validate Your HTML:** Use an online HTML validator to check your code for errors.

#### Step 9: Reflect and Document
1. **Write a Summary:** Document what you learned from this exercise.
2. **Challenges Faced:** Note any challenges you encountered and how you solved them.

#### Step 10: Further Learning
1. **Read More:** Explore more advanced HTML topics and tags.
2. **Practice Regularly:** Keep practicing by creating more complex web pages.

---

Experiment and explore additional HTML elements and attributes to gain a deeper understanding of web page structure.

---

Let's expand on the initial exercise and include more detailed explanations and additional steps to deepen the understanding of HTML for new MEAN stack learners. This enhanced exercise will cover not only the practical steps but also the underlying concepts and theories.

#### Introduction to HTML
- **What is HTML?** HTML (HyperText Markup Language) is the standard language used to create and design documents on the World Wide Web. HTML describes the structure of a web page semantically and originally included cues for the appearance of the document.

- **HTML Elements:** An HTML element usually consists of a start tag and an end tag, with the content inserted in between: `<tagname>Content goes here...</tagname>`. Elements are the building blocks of HTML pages.

- **HTML Tags:** Tags are used for creating elements. For example, `<p>` is a tag for paragraph elements.

#### Step 1: Setting Up Your Environment
- **Tools Required:** Code Editor (like Visual Studio Code), Web Browser.
- **Creating a File:** `index.html` in a project folder named `html_basics`.

#### Step 2: Basic Structure of an HTML Document
- **Doctype Declaration:** `<!DOCTYPE html>` is essential to let the browser know that this is an HTML5 document.
- **Root Element:** `<html>` is the root element that will contain all other HTML elements.

#### Step 3: Head and Body Sections
- **The `<head>` Element:** Contains meta-information like the title of the document, character set, stylesheets, and more. 
  - **Title:** The `<title>` tag defines the title of the document, which is shown in the browser's title bar or tab.
- **The `<body>` Element:** This is where all the visible content of your HTML document goes, like text, images, links, etc.

#### Step 4: Adding Content to the Body
- **Headings:** HTML offers six levels of headings (`<h1>` to `<h6>`). `<h1>` is the most important (usually the main heading), and `<h6>` is the least.
- **Paragraphs:** The `<p>` tag defines a paragraph. Browsers automatically add some space (margin) before and after each paragraph.

#### Step 5: Creating a List
- **Unordered Lists (`<ul>`):** Used for lists where the order doesn't matter. Each item in the list is marked with a bullet point.
- **List Item (`<li>`):** Represents an item in a list.

#### Step 6: Adding Links and Images
- **Hyperlinks (`<a>`):** The `<a>` tag defines a hyperlink, which is used to link from one page to another. The most important attribute of the `<a>` element is the `href` attribute, which indicates the link's destination.
- **Images (`<img>`):** The `<img>` tag is used to embed an image in an HTML page. The `src` attribute specifies the path to the image, and the `alt` attribute provides alternative information for an image if a user for some reason cannot view it (due to slow connection, an error in the `src` attribute, or if the user uses a screen reader).

#### Step 7: Viewing Your Web Page
- **Testing:** Open your `index.html` file in a web browser to view your work.

#### Step 8: Experimentation and Practice
- **Modification and Addition:** Encourage playing around with different tags and attributes.
- **HTML Validation:** Emphasize the importance of clean, valid HTML. Use tools like the W3C Markup Validation Service.

#### Step 9: Reflect and Document
- **Understanding HTML:** Write about how HTML is not a programming language, but a markup language intended to store and transport data.
- **Challenges:** Document any difficulties encountered and how they were overcome.

#### Step 10: Further Learning
- **Advanced Topics:** Look into forms, tables, semantic HTML5 elements like `<article>`, `<section>`, `<header>`, `<footer>`, etc.
- **Consistent Practice:** Build different types of web pages to get comfortable with HTML.

---

Understanding these concepts is crucial for any web developer, especially those starting with the MEAN stack, as it lays the groundwork for more advanced topics in web development.

**Comment by dmpatil1990** [comment]
<img width="527" alt="GE-229" src="https://github.com/akash-coded/mern/assets/135525592/8d3b1ed6-2da0-499c-bbcc-ca77120d56ec">

**Reply by monalip92** [reply]
![firstjpg](https://github.com/akash-coded/mern/assets/144552652/b5c7face-c935-452c-9d5b-5d280dcdad2f)

**Comment by priyeranjan** [comment]
<img width="960" alt="image" src="https://github.com/akash-coded/mern/assets/7481567/ee299157-9674-4cc0-a9d6-8e82a9539ea7">

**Reply by sidchakravarty** [reply]
Very nice!

**Comment by sidchakravarty** [comment]
![Assignment - 1](https://github.com/akash-coded/mern/assets/15659742/1cd52fb5-64dd-4662-9819-e30a870553f6)

**Comment by vpervela** [comment]
<img width="463" alt="Screen Shot 2023-11-18 at 8 59 07 AM" src="https://github.com/akash-coded/mern/assets/101532142/d0d0df1d-69e8-4a42-a216-90ee3a491dd5">

**Reply by Balajik87-hub** [reply]
<!DOCTYPE html>
<html>
<head>
    <meta charset='utf-8'>
    <meta http-equiv='X-UA-Compatible' content='IE=edge'>
    <title>Document</title>
    <meta name='viewport' content='width=device-width, initial-scale=1'>
    <link rel='stylesheet' type='text/css' media='screen' href='main.css'>
    <script src='main.js'></script>
</head>
<body>
    <h1> First HTML webdesign </h1>
    <h2> First HTML webdesign </h2>
    <h3> First HTML webdesign </h3>
    <h4> First HTML webdesign </h4>
    <h5> First HTML webdesign </h5>
    <h6> First HTML webdesign </h6>

    <p> creating first html page</p>

    <ul>
        <li>HTML1</li>
        <li>HTML2</li>
        <li>HTML3</li>
        <li>HTML4</li>
    </ul>
    <a href='www.google.com'></a>
    <img src="" alt="">
</body>
</html>

**Comment by VarshiniBS** [comment]
![Screenshot (686)](https://github.com/akash-coded/mern/assets/87222601/671af81f-5a68-4812-aec8-53345285f7a8)

**Comment by SwetaSaha1** [comment]
![Screenshot 2023-11-18 at 10 38 26 PM](https://github.com/akash-coded/mern/assets/127398967/459073f8-209f-4b80-b0e4-eaa48533b3c2)

**Comment by Lellagopi143** [comment]
![Screenshot 2023-11-18 223054](https://github.com/akash-coded/mern/assets/151382935/37dbe4b4-44b2-4ec6-add1-0d74197d67ae)

**Comment by mygit-shirinA** [comment]
![image](https://github.com/akash-coded/mern/assets/151388915/02eb6319-9d4b-45ef-9095-e0349c774354)

**Reply by bmuvara** [reply]
![exercise1](https://github.com/akash-coded/mern/assets/14073436/01becd08-7db2-4003-a62a-5d8ab3405d47)

**Comment by Dhanalakshmi-S-23** [comment]
![229](https://github.com/akash-coded/mern/assets/151384783/6fd0d7d1-3698-478c-8165-a2afe949fc95)

**Comment by santhoshraj123** [comment]
![IMG20231118225519.jpg](https://github.com/akash-coded/mern/assets/827096/0f98946b-5482-44a8-9033-b63e4efcbac0)

**Comment by ghost** [comment]
![Screenshot (255)](https://github.com/akash-coded/mern/assets/135525365/2154763f-1e86-4f14-80bb-fa24c1949721)

**Comment by RichardChhetri** [comment]
![image (2)](https://github.com/akash-coded/mern/assets/74686643/30c49ba0-1b24-4c0c-9153-b8fb88276386)

**Comment by arvid-mvo** [comment]
![upload_1](https://github.com/akash-coded/mern/assets/101291172/72371b8b-4fae-4c67-b54f-8445102dd5b7)

**Comment by bacchus19408** [comment]
<img width="644" alt="Screen Shot 2023-11-18 at 6 43 33 PM" src="https://github.com/akash-coded/mern/assets/4041048/f56731a5-6764-4db2-8d33-51180ebc5662">

**Comment by ujjwal432** [comment]
<img width="1091" alt="Html Exercise #229" src="https://github.com/akash-coded/mern/assets/61551049/a6740b32-c81b-4a3d-aaf6-171b16fb7c20">

**Comment by VIJAY1999v** [comment]
![Screenshot (326)](https://github.com/akash-coded/mern/assets/126997615/13a7f6ff-6fd6-4543-80f7-d99aae4bb9de)

**Comment by AbiramiMadhusudan28** [comment]
<img width="1470" alt="Screenshot 2023-11-19 at 12 13 32 AM" src="https://github.com/akash-coded/mern/assets/151383196/d31c0538-635a-4969-87cb-97ffb081f68d">

**Comment by JJismi** [comment]
<img width="960" alt="html " src="https://github.com/akash-coded/mern/assets/133745306/51edd126-5161-403b-b115-476f7deb2cce">

**Comment by Dipalivrma** [comment]
![image](https://github.com/akash-coded/mern/assets/119152780/a8395b8c-37ee-4344-bfb6-dbc835ed1c42)

**Comment by AnshuNag** [comment]
![image](https://github.com/akash-coded/mern/assets/151382909/858d76c6-e9c3-49fc-bfea-6f0ff7204010)

**Comment by aslamcsengineer** [comment]
![Screenshot 2023-11-19 185423](https://github.com/akash-coded/mern/assets/151382329/2c81a7cc-b391-493b-80ff-c88546d1895a)

**Comment by HFahimm** [comment]
![Screenshot 2023-11-19 191248](https://github.com/akash-coded/mern/assets/151382359/54149d7e-0064-476a-b660-c281c05825bf)

**Comment by SimplySneha27** [comment]
![html_index](https://github.com/akash-coded/mern/assets/151423861/674b75ca-ec8e-4a87-82bf-8686cc66c1ed)

**Comment by rosekthomas** [comment]
![test](https://github.com/akash-coded/mern/assets/8777388/338d0098-70be-467f-8d52-c3c609590521)

**Comment by Jayashreewaghmare** [comment]
![Screenshot 2023-11-19 202633](https://github.com/akash-coded/mern/assets/127091548/68a643ed-6482-4c47-8d42-8d6ea181bb7e)

**Comment by padma1998lela** [comment]
![Screenshot 2023-11-20 150007](https://github.com/akash-coded/mern/assets/62737217/cb5fea61-3ff3-4a9b-bc3f-b5a9e7f557f4)

**Comment by Sunil-nani** [comment]
![Uploading Screenshot (2).png…]()

**Comment by Umarani1711** [comment]
<img width="629" alt="Html First Page" src="https://github.com/akash-coded/mern/assets/78922312/8c7c833b-4fca-43e3-be7b-e42afd65b502">

**Comment by DeepakrajRavi** [comment]
![Screenshot 2023-11-23 115552](https://github.com/akash-coded/mern/assets/151645622/9c2d48b3-6b78-45ae-b25c-41c6fe6aa2b5)

**Comment by PUMAGrisen** [comment]
![Namnlös](https://github.com/akash-coded/mern/assets/151846056/3ae71794-fda5-4c8e-8a4c-51f4fb9c5c91)

**Comment by Geethanjali5** [comment]
![IMG-20231123-WA0000](https://github.com/akash-coded/mern/assets/147892641/f9a075a7-e496-4d0e-9346-cae3fc986fb9)

**Comment by ankitarawat1310** [comment]
![image](https://github.com/akash-coded/mern/assets/34696259/86ab365f-0be9-4894-b2cf-59303f145b71)

**Comment by SwetaSaha1** [comment]
![Screenshot 2023-11-30 at 10 21 11 PM](https://github.com/akash-coded/mern/assets/127398967/5faa3f6d-684c-48d6-88d1-f4c05be0d85e)
